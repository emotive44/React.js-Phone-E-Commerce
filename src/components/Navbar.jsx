import React from 'react';
import { Link } from 'react-router-dom';
import logo from '../logo.svg';
import ButtonContainer from './styles/Button';

class Navbar extends React.Component {
    render() {
        return(
            <nav className="navbar navbar-expand-sm navbar-dark px-sm-5 bg-primary">
                <Link to='/'>
                    <img src={logo} alt="store" className="navbar-brand" />
                </Link>
                <ul className="navbar-nav align-items-center">
                    <li className="nav-item ml-5">
                        <Link to='/' className="nav-link">Products</Link>
                    </li>
                </ul>
                <Link to='/cart' className="ml-auto">
                    <ButtonContainer>
                        <span className="mr-2">
                            <i className="fas fa-cart-plus" />
                        </span>
                        My cart
                    </ButtonContainer>
                </Link>
            </nav>
        )
    }
}

export default Navbar;
