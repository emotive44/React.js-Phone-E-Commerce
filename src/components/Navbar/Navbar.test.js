import React from 'react';
import Navbar from './Navbar';
import renderer from 'react-test-renderer';

jest.mock('react-router-dom', () => ({
    Link: 'Link'
}));

describe('Navbar component', () => {
    test('should reder correctly', () => {
        const component = renderer.create(<Navbar />);
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    })
})