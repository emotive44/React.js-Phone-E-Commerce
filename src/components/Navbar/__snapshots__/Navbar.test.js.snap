// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar component should reder correctly 1`] = `
<nav
  className="sc-AxirZ hPnYIg navbar navbar-expand-sm navbar-dark px-sm-5 "
>
  <Link
    to="/"
  >
    <img
      alt="store"
      className="navbar-brand"
      src="logo.svg"
    />
  </Link>
  <ul
    className="navbar-nav align-items-center"
  >
    <li
      className="nav-item ml-5"
    >
      <Link
        className="nav-link"
        to="/"
      >
        Products
      </Link>
    </li>
  </ul>
  <Link
    className="ml-auto"
    to="/store"
  >
    <button
      className="sc-AxjAm IwjNC"
    >
      <span
        className="mr-2"
      >
        <i
          className="fas fa-cart-plus"
        />
      </span>
      My cart
    </button>
  </Link>
</nav>
`;
